src/HttpClientFactory/Polly/test/PolicyHttpMessageHandlerTest.cs

```diff
@@ -98,6 +99,54 @@ public async Task SendAsync_DynamicPolicy_PolicyTriggers_CanReexecuteSendAsync()
             Assert.Same(expectedRequest, policySelectorRequest);
         }
 
+        [Fact]
+        public async Task SendAsync_StaticPolicy_PolicyTriggers_CanReexecuteSendAsync_FirstResponseDisposed()
+        {
+            // Arrange
+            var policy = HttpPolicyExtensions.HandleTransientHttpError()
+                .RetryAsync(retryCount: 1);
```

Comment:
Assuming I understand what this is doing correctly, the original issue report seems to indicate that it happens after two retries. I assume that that was just a scenario specific detail and validating on 1 retry is enough for this case?

Reply:
Right, in their scenario it took two leaks to notice the problem because that's how many resources they had available, two connections. The solution is to always dispose the previous response, it doesn't matter how many retries there are.